src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -58,6 +60,32 @@ public class LinkedInAuthenticationHandler : OAuthHandler<LinkedInAuthentication
             }
 
             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            if (Options.Fields.Contains(LinkedInAuthenticationConstants.EmailAddressField))
```

Comment:
Should this `Contains()` check maybe be case-insensitive?

Reply:
Good point too! I am now using the `HashSet(System.Collections.Generic.IEqualityComparer<T> comparer)` constructor to initialize `Fields` property